### ISSUE TITLE
Multilevel API

### ISSUE THREAD

**Issue by rofinn** [mateuszbaran/CovarianceEstimation.jl#10]
It might be nice to support a lower level `fit` and `transform` API on top of just `cov`. This would allow folks to pass in an existing covariance matrix that they'd like to apply an estimation method to (e.g., a weighted covariance).

**Comment by mateuszbaran** [comment]
I don't quite understand how would that work. Implemented formulas for covariance shrinkage assume normal covariance and a specific shrinkage target. Wouldn't using weighted covariance lead to wrong shrinkage coefficients?

**Comment by mateuszbaran** [comment]
StatsBase.jl has its own approach to weights and it looks nicer. I think it would be good to have a similar design here.

**Comment by nickrobinson251** [comment]
Supporting weights, following the StatsBase API of `cov(ce::CovarianceEstimator, X::AbstractMatrix, w::AbstractWeights)`, would be really useful :)

**Comment by mateuszbaran** [comment]
Yes, it would be very useful, but I still don't know if it's even possible to have generic implementation for all types of weights. Frequency weights should be relatively easy to support though. What kind of weights do you need?

**Comment by nickrobinson251** [comment]
I believe `AnalyticWeights`. But also I am not sure weighted Lediot-Wolf is possible (I was hoping it was)

**Comment by mateuszbaran** [comment]
Unfortunately, I don't know how to modify Ledoit-Wolf to make it work with `AnalyticWeights`. If you find a paper about it, I will definitely like to see it.

**Comment by nickrobinson251** [comment]
My colleague tells me it is not possible (or at least the proof in the LW paper breaks down when weights are added) ...but i've not done the working myself nor seen a paper on it.

**Comment by rofinn** [comment]
NOTE: a potential work around could be to collect weighted sample first and then estimate the covariance from that.
